analyses/doublet-detection/scripts/01a_detect-doublets.R

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env Rscript
+
+library(SingleCellExperiment)
+library(optparse)
+
+# Setup ------
+
+# Parse options
+option_list <- list(
+  make_option(
+    c("-c", "--cores"),
+    type = "character",
+    help = "Number of cores to use during scDblFinder inference."
+  )
+)
+opts <- parse_args(OptionParser(option_list = option_list))
+
+
+# Define directories
+module_base <- file.path(
+  rprojroot::find_root(rprojroot::is_git_root),
+  "analyses",
+  "doublet-detection"
+)
+data_dir <- file.path(module_base, "scratch", "benchmark_datasets")
+result_dir <- file.path(module_base, "results", "benchmark_results")
+fs::dir_create(result_dir)
+
+
+# Functions -----------
+
+#' Run scDblFinder on an SCE object
+#'
+#' @param sce SCE object to run scDblFinder on
+#' @param cores Number of cores to specify to BiocParallel::MulticoreParam(
+#' @param sample_var Required for multiplexed data, the SCE colData variable that indicates sample of origin
+#' @param random_seed Required for multiplexed data, a random seed to ensure reproducibility when running on multiplexed data.
```

Comment:
Why do we only want a seed for multiplexed data?

Reply:
I linked this below on line 56: see section 1.5.6  here https://bioconductor.org/packages/3.19/bioc/vignettes/scDblFinder/inst/doc/scDblFinder.html#usage. Apparently `set.seed()` in the R script is enough for single samples only, and our main circumstance for that would be multiplexed data (that said, we don't actually _perform_ demultiplexing currently to be able to identify this, but who knows maybe it'll happen during OpenScPCA!). Either way, I should update this doc to say "multiple samples" instead of "multiplexed data"

Reply:
Worth noting I was indeed missing a seed more generally though, and in the python script(!)